python/ray/gateway.py

```diff
@@ -0,0 +1,110 @@
+import click
+from flask import Flask, request, send_file
+import io
+import pickle
+import pyarrow
+from pyarrow import plasma as plasma
+import sys
+
+app = Flask(__name__)
+ctx = pyarrow.default_serialization_context()
+
+@app.route('/', methods=['GET', 'POST'])
+def index():
+    """TODO (dsuo): Add comments.
+
+    TODO (dsuo): Maybe care about batching
+    TODO (dsuo): REST is nice conceptually, but probably too much overhead
+    TODO (dsuo): Move logic to C++ (keep as head node process)
+
+    NOTE: We do an extra serialize during get and extra deserialize during
+        put.
+    """
+    if request.method == 'POST':
+        object_id = request.files['object_id'].read()
+        data = request.files['value'].read()
+        value = ctx.deserialize(data)
+
+        plasma_client.put(
+            value,
+            object_id=plasma.ObjectID(object_id),
+            memcopy_threads=12,
+            serialization_context=ctx)
+
+        return object_id, 402
+    elif request.method == 'GET' and 'object_ids' in request.args:
+        object_ids = [plasma.ObjectID(bytes.fromhex(object_id)) for object_id in \
+            request.args['object_ids'].split(",")]
+
+        # Fetch remote objects
+        # TODO (dsuo): maybe care about batching
+        plasma_client.fetch(object_ids)
+
+        # Get local objects
+        # TODO (dsuo): Handle object not available
+        results = plasma_client.get(
```

Comment:
If we want to just get the bytes and not actually deserialize the object, then I think we can use `plasma_client.get_buffers` or something like that.

Reply:
Yeah, thanks for that. In the current code, we do the extra serialization / deserialization.